stack-graphs/src/graph.rs

```diff
@@ -1442,9 +1449,10 @@ pub struct StackGraph {
     pub(crate) nodes: Arena<Node>,
     pub(crate) source_info: SupplementalArena<Node, SourceInfo>,
     node_id_handles: NodeIDHandles,
-    outgoing_edges: SupplementalArena<Node, SmallVec<[OutgoingEdge; 8]>>,
+    outgoing_edges: SupplementalArena<Node, SmallVec<[OutgoingEdge; 4]>>,
+    incoming_edges: SupplementalArena<Node, u32>,
```

Comment:
Given the other discussion about relaxing the stitching restriction, you might consider not tracking the actual in-degree of each node, but instead tracking something like

``` rust
#[repr(u8)]
enum Degree {
  Zero,
  One,
  Multiple,
}
```

That would cut down 4× on the storage space of this new arena and would eventually present a cleaner API if we decide to track this info through the storage layer

Reply:
That's a nice idea! In cd58abbff3d128fc480f7f016c3ac1ed9d01bf04.